flybywiresim/aircraft
# feat(a380x/fms): FMC reset through overhead reset panel

## Summary of Changes

Enable resetting FMCs through overhead reset panel. Lays ground work for future system resets. 

- [x] Add XML templates for overhead reset panel buttons for FMCs
- [x] Add logic for FMS re-initialization when pulling C/Bs or rather reset P/Bs
- [x] Display "FMS PAGE NOT AVAIL" and "MAP NOT AVAIL" when appropriate

## Screenshots (if necessary)


![image](https://github.com/user-attachments/assets/54ff8d45-9d2b-4da1-98ad-664f9d684646)

Disclaimer: Even when the ND shows "MAP NOT AVAIL", the route is still shown. Either it's the new BACKUP TRAJ feature, or I simply didn't refactor the ND yet (out of scope for this PR, needs larger flight guidance/AFS CP adaptions likely):
![image](https://github.com/user-attachments/assets/97625a57-6b34-4818-9645-7b88442066f8)



## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions

1. Set up FMS using SOP
2. Pull FMC-A, FMC-B and FMC-C reset push buttons (on overhead panel), FMS should be reset as soon as all three buttons are pulled out - but not earlier
3. Try resetting or failing (via EFB) different combinations of FMCs, and play around with the FMS switching knob. The MFD system selector ("FMS x") should behave as expected, i.e. display FMS 1, FMS 2, FMS 1-C or FMS 2-C depending on the configuration. 


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
